pandas/core/arrays/masked.py

```diff
@@ -229,7 +240,7 @@ def fillna(
             if method is not None:
                 func = missing.get_fill_func(method, ndim=self.ndim)
                 npvalues = self._data.T
-                new_mask = mask.T
+                new_mask = mask.to_numpy().T
```

Comment:
can we avoid this conversion?  maybe feasible-but-out-of-scope?

Reply:
Are you asking about avoiding the `T` even in the current code? Or implementing support for transposition in the BitmaskArray directly?

Reply:
> Or implementing support for transposition in the BitmaskArray directly?

That one

Reply:
The two options for that are:
  1. Physically transpose the layout of the bits in memory
  2. Keep the bits the same and just use state management to resolve indexes to a different address when transposed

I don't know which of these NumPy does. Maybe a combination of both.  If we want point 1, we also need to find / develop a good algorithm for it

It's possible just requires some dedicated time just to that topic